core/modules/xrdlog/XrdLogging.cc

```diff
@@ -0,0 +1,74 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/// Implement logging hook to route xrootd/cmsd messages to our logger.
+/// This source file has no header file.
+
+// Class header
+
+// System headers
+#include <string>
+
+// Third party headers
+#include "XrdSsi/XrdSsiLogger.hh"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+
+/******************************************************************************/
+/*                L o g g i n g   I n t e r c e p t   H o o k                 */
+/******************************************************************************/
+  
+namespace {
+
+const char* origin;
```

Comment:
Well, if I move it to another place it will likely will have to become a class of its own so the origin can be properly set by either the czar or the logging plugin. It certainly cannot be anonymous anymore.